### DISCUSSION TITLE
window-height setting only works if it's more than 22

### DISCUSSION THREAD

**Discussion by sidesteps** [ghostty-org/ghostty#3702] [Q&A] [answered]
On linux xfce with standard composing windows manager (wezterm, xfce-terminal respects those settings no problem)

```
window-width=196
window-height=16
```

but ghostty always starts at height 22.

```
$tput lines 
->22
```

If I set window-height to more than 22, then the setting is respected.

**Comment by asheriif** [comment]
I'm experiencing the exact same issue on MacOS. 

```
window-width = 86
window-height = 17
```

yet ghostty always starts with an initial height of 24 for me. If I set it to anything > 24, it's respected.

**Reply by sidesteps** [reply]
> I'm experiencing the exact same issue on MacOS.
> 
> ```
> window-width = 86
> window-height = 17
> ```
> 
> yet ghostty always starts with an initial height of 24 for me. If I set it to anything > 24, it's respected.

Well the good thing is it's not a bug - it's a discussion.

**Reply by liby** [reply]
I was unable to reproduce this issue. Instead, when I set `window-height=x`, `tput lines` always outputs x+1.

<img width="724" alt="image" src="https://github.com/user-attachments/assets/3f0849f5-68ff-47df-a084-7a02103fc3ba" />

**Reply by mitchellh** [reply]
Yeah note there are off by one issues on height (might even be off by more than that depending on font size) as documented: https://ghostty.org/docs/config/reference#window-height If anyone knows how to fix those I'd love to know how. 😄

**Reply by liby** [reply]
I've done some additional testing regarding the minimum window size behavior:
<img width="639" alt="image" src="https://github.com/user-attachments/assets/4042a7d4-45be-4277-b36b-6f9f99d777a3" />

Interestingly, I can no longer reproduce my previous observation where "when setting window-height=x, tput lines always outputs x+1". The behavior seems to have changed.

**Comment by mitchellh** [comment, marked as answer]
Dupe #4653
